### PR TITLE
Migrated App2 and App3 tests to MariaDB

### PR DESCRIPTION
**Pull request by zwall-bp**
Before approving, this PR is assuming that the [osl-nginx](https://github.com/osuosl-cookbooks/osl-nginx/pull/44) cookbook has been updated.

### CODE REVIEW

**Comment by ramereth** [comment]
@sectopodwreck let's wait until #104 is resolved and merged

**Comment by ramereth** [comment]
@zwall-bp please rebase and resolve the conflicts

File: test/cookbooks/app_test/recipes/app2.rb
Comment (ramereth) [inline test/cookbooks/app_test/recipes/app2.rb:21]: Hmm, it might be a good idea to add this to the `osl_mysql_test` resource.

File: test/cookbooks/app_test/recipes/app2.rb
Comment (ramereth) [inline test/cookbooks/app_test/recipes/app2.rb:1]: Why is this still needed with the `osl_mysql_test` resource?

**Review by ramereth — CHANGES_REQUESTED** [review]

File: test/cookbooks/app_test/recipes/app2.rb
Comment (zwall-bp) [reply test/cookbooks/app_test/recipes/app2.rb]: I forgot that I have removed this attribute from the `kitchen.yml`, so this is no longer required. It's been removed.

File: test/cookbooks/app_test/recipes/app2.rb
Comment (zwall-bp) [reply test/cookbooks/app_test/recipes/app2.rb]: Implemented in the latest PR to `osl-mysql`.

**Comment by zwall-bp** [comment]
This PR assumes that another PR for `osl-mysql` has been merged.

https://github.com/osuosl-cookbooks/osl-mysql/pull/79

**Review by ramereth — CHANGES_REQUESTED** [review]
The postrun test recipes need some adjustment as there's an `edit_resource` that likely needs to be removed and is failing the chef run.

**Comment by ramereth** [comment]
@zwall-bp I'm still seeing InSpec test failures on a fresh run. Can you please look into those?

**Comment by zwall-bp** [comment]
Seems like the merge update removed my changes, let me quickly patch that.

**Comment by zwall-bp** [comment]
> @zwall-bp I'm still seeing InSpec test failures on a fresh run. Can you please look into those?

Which suites are raising errors? I just tested all of the suites again without any issues.

**Comment by ramereth** [comment]
I'm having issues with the app3 suite:
```
     ×  HTTP GET on http://127.0.0.1:8086/ status is expected to eq 200
     
     expected: 200
          got: nil
     
     (compared using ==)

     ×  HTTP GET on http://127.0.0.1:8086/ body is expected to match "<title>Etherpad</title>"
     expected nil to match "<title>Etherpad</title>"
     ×  HTTP GET on http://127.0.0.1:8086/small_list status is expected to eq 200
     
     expected: 200
          got: nil
     
     (compared using ==)
     ×  Command: `docker exec etherpad-snowdrift.osuosl.org env` stdout is expected to match "DB_TYPE=mysql"
     expected "" to match "DB_TYPE=mysql"
     ×  Command: `docker exec etherpad-snowdrift.osuosl.org env` stdout is expected to match "DB_HOST=10.1.100.185"
     expected "" to match "DB_HOST=10.1.100.185"
     ×  Command: `docker exec etherpad-snowdrift.osuosl.org env` stdout is expected to match "DB_NAME=etherpad_sd"
     expected "" to match "DB_NAME=etherpad_sd"
     ×  Command: `docker exec etherpad-snowdrift.osuosl.org env` stdout is expected to match "DB_USER=etherpad_sd"
     expected "" to match "DB_USER=etherpad_sd"
     ×  Command: `docker exec etherpad-snowdrift.osuosl.org env` stdout is expected to match "DB_PASS=super_safe"
     expected "" to match "DB_PASS=super_safe"
     ×  Command: `docker exec etherpad-snowdrift.osuosl.org env` stdout is expected to match "ADMIN_PASSWORD=adminpasswd"
     expected "" to match "ADMIN_PASSWORD=adminpasswd"
```

**Comment by zwall-bp** [comment]
Ah, I remember that error, it had to do with both the firewall and there was a user who didn't have permissions to connect from outside localhost. These should have been patched quite a while ago though.

Let me run another test for app3 on my end to see if those pop up again.

**Review by ramereth — APPROVED** [review]
Merge as-is. It seems to be working for me now :shrug:

**Pull request merged**
